api/authorization-api-1_0.md

```diff
@@ -67,187 +67,292 @@ normative:
 
-type:
-: OPTIONAL. The type of the resource. Its value is a `string` that specifies the type of the resource
+##### Device ID {#subject-device-id}
+The Device Identifier of the Subject, identified by a `device_id` field, whose value is a string representation of the device identifier.
 
+The following is a non-normative example of a subject which adds the `device_id` attribute:
 
-#### Example (non-normative)
+~~~ json
+{
+  "type": "user",
+  "id": {
+    "format" : "email",
+    "email": "alice@acmecorp.com"
+  },
+  "department": "Sales",
+  "ip_address": "172.217.22.14",
+  "device_id": "8:65:ee:17:7e:0b"
+}
+~~~
+{: #subject-device-id-example title="Example Subject with Device ID"}
+
+### Resources {#resources}
+A Resource is the target of an access request. It is a JSON ({{RFC8259}}) object that is constructed similar to a Subject entity.
 
-The following is a non-normative example of a Resource:
+`type`:
+: REQUIRED. A `string` value that specifies the type of the Resource.
 
-~~~json
+`id`:
+: OPTIONAL. The unique identifier of the Resource, scoped to the `type`, within the scope of the PEP. If specified, the value MUST follow the format specified by the `Subject Identifiers for Security Event Tokens` specification {{RFC9493}}.
+
+#### Example (non-normative)
+The following is a non-normative example of a Resource containing a Subject Identifier in the Opaque Identifier Format:
+
+~~~ json
 {
-  "id": "123",
   "type": "book",
-  "libraryRecord":{
-    "title": "ABAC's new hit",
+  "id": {
+    "format": "opaque",
+    "value": "123"
+  },
+  "library_record":{
+    "title": "AuthZEN in Action",
     "isbn": "978-0593383322"
   }
 }
 ~~~
-{: #resourceexample title="Example Resource"}
+{: #resource-example title="Example Resource"}
 
 ### Actions {#actions}
-An action is the type of access that the requester intends to perform. There are common actions defined herein, or the action may be custom, which could be specific to the application being accessed or shared across applications but not listed in the common actions below. Action is a JSON ({{RFC8259}}) object that is constructed just like a Subject or Resource object.
+An Action is the type of access that the requester intends to perform.
+
+Action is a JSON ({{RFC8259}}) object that contains at least a `name` field.
+
+`name`:
+: REQUIRED. The name of the Action.
+
+The following is a non-normative example of an action:
+
+~~~ json
+{
+  "name": "can_read"
+}
+~~~
+{: #action-example title="Example Action"}
+
+#### Common Action Values
+Since many services follow a Create-Read-Update-Delete convention, a set of common Actions are defined. That said, an Action may be custom, which could be specific to the application being accessed or shared across applications but not listed in the common Actions below.
+
+The following common Actions are defined:
```

Comment:
I think the values are not "ABAC" enough. Why specify values in the first place? And why not just use the verbs e.g. view, edit, update, access rather than an artificial can_read? Also, why use underscore?

Reply:
These are meant to read as permissions.  The convention that ReBAC systems use to differentiate permissions from relations are that permissions start with `can_` (and are not directly assignable).

I personally don't think the "common Actions" are that useful, but they were in the current draft and I believe there was some consensus that we wanted common values.